packages/core/client/src/spin/index.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { Spin as SpinComponent } from 'antd';
+import { css } from '@emotion/css';
+
+interface IProps {
+  spinning?: boolean;
+  children?: React.ReactNode;
+}
+
+export function Spin({ spinning, children }: IProps) {
+  return <SpinComponent 
+    className={css`
+      height: 100%;
+      width: 100%;
+      justify-content: center;
+      display: flex;
+      align-items: center;
+    `}
+    spinning={spinning}
+  >{children}</SpinComponent>;
+};
```

Comment:
最好不要把重写 antd 的组件直接放在 src 下，如果后期需要对其它组件进行重写按照现在这样设计的话这个目录会很难看，建议单独弄一个目录来存放重写 anrd 的组件和自定义的组件，可以在 src 下单独弄一个 components 目录吧

Reply:
另外这个组件最好做一下兼容，就是得支持 Spin 本身支持的属性

```suggestion
import React from 'react';
import { Spin as SpinComponent } from 'antd';
import type { SpinProps } from 'antd';
import { css } from '@emotion/css';

export function Spin({ children, ...otherProps }: SpinProps) {
  return <SpinComponent
    className={css`
      height: 100%;
      width: 100%;
      justify-content: center;
      display: flex;
      align-items: center;
    `}
    {...otherProps}
  >{children}</SpinComponent>;
};
```